apiserver/facades/agent/caasagent/register.go

```diff
@@ -32,26 +32,26 @@ func newStateFacadeV2(ctx facade.ModelContext) (*FacadeV2, error) {
 	}
 
 	resources := ctx.Resources()
-	model, err := ctx.State().Model()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
+
+	authFunc := common.AuthFuncForTag(names.NewModelTag(string(ctx.ModelUUID())))
 
 	domainServices := ctx.DomainServices()
 
+	// Create functions that don't use state.State but use domain services instead
 	cloudSpecAPI := cloudspec.NewCloudSpecV2(
 		resources,
-		cloudspec.MakeCloudSpecGetterForModel(ctx.State(), domainServices.Cloud(), domainServices.Credential(), domainServices.Config()),
-		cloudspec.MakeCloudSpecWatcherForModel(ctx.State(), domainServices.Cloud()),
-		cloudspec.MakeCloudSpecCredentialWatcherForModel(ctx.State()),
-		cloudspec.MakeCloudSpecCredentialContentWatcherForModel(ctx.State(), domainServices.Credential()),
-		common.AuthFuncForTag(model.ModelTag()),
+		cloudspec.MakeCloudSpecGetterForModel(nil, domainServices.Cloud(), domainServices.Credential(), domainServices.Config()),
+		cloudspec.MakeCloudSpecWatcherForModel(nil, domainServices.Cloud()),
+		cloudspec.MakeCloudSpecCredentialWatcherForModel(nil),
+		cloudspec.MakeCloudSpecCredentialContentWatcherForModel(nil, domainServices.Credential()),
+		authFunc,
 	)
+
 	return &FacadeV2{
 		CloudSpecer:        cloudSpecAPI,
 		ModelConfigWatcher: commonmodel.NewModelConfigWatcher(domainServices.Config(), ctx.WatcherRegistry()),
 		ControllerConfigAPI: common.NewControllerConfigAPI(
-			ctx.State(),
```

Comment:
We want to leave this in here for the moment please. The goal of this task is remove just the calls mongo state for model's. We don't have a migration plan for this yet.

Reply:
fixed.